kernel/multiple_heaps/src/lib.rs

```diff
@@ -111,20 +112,16 @@ pub fn switch_to_multiple_heaps() -> Result<(), &'static str> {
 /// Allocates pages from the given starting address and maps them to frames.
 /// Returns the new mapped pages or an error if the heap memory limit is reached.
 fn create_heap_mapping(starting_address: VirtualAddress, size_in_bytes: usize) -> Result<MappedPages, &'static str> {
-    if (starting_address.value() + size_in_bytes) >  (KERNEL_HEAP_START + KERNEL_HEAP_MAX_SIZE) {
-        return Err("Heap memory limit has been reached");
-    }
-
     let kernel_mmi_ref = get_kernel_mmi_ref().ok_or("create_heap_mapping(): KERNEL_MMI was not yet initialized!")?;
     let mut kernel_mmi = kernel_mmi_ref.lock();
 
     let mut frame_allocator = get_frame_allocator_ref()
         .ok_or("create_heap_mapping(): couldnt get FRAME_ALLOCATOR")?
         .lock();
 
-    let pages = PageRange::from_virt_addr(starting_address, size_in_bytes);
-    let heap_flags = HEAP_FLAGS;
-    let mp = kernel_mmi.page_table.map_pages(pages, heap_flags, frame_allocator.deref_mut())?;
+    let pages = page_allocator::allocate_pages_by_bytes_at(starting_address, size_in_bytes)
```

Comment:
The heap will fail here if the page at this address is not available. It's unlikely since the chunk which contains these addresses is after the kernel text sections. It can be made more generic where the heap is can request any page as long as it's 8k aligned.